inc/drivers/MicroBitPin.h

```diff
@@ -95,8 +95,8 @@ class MicroBitPin : public MicroBitComponent
 {
     // The mbed object looking after this pin at any point in time (untyped due to dynamic behaviour).
     void *pin;
-
     PinCapability capability;
+    uint8_t pullMode;
```

Comment:
This is going to add (probably) 4 bytes of RAM to each pin.  That means taking about 80 bytes when all ~20 pins are created.